sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -180,3 +182,214 @@ export function getAssociatedReceiverName(
   }
   return receiverName;
 }
+
+/**
+ *  @ignore
+ * Helper utility to retrieve `string` value from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getStringOrUndefined(value: any): string | undefined {
+  if (value != undefined) {
+    return value.toString();
+  } else {
+    return undefined;
+  }
+}
+
+/**
+ *  @ignore
+ * Helper utility to retrieve `integer` value from given string,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getIntegerOrUndefined(value: any): number | undefined {
+  if (value != undefined) {
+    try {
+      return parseInt(value.toString());
+    } catch (err) {
+      throw new TypeError(`${value} expected to be a number or undefined`);
+    }
+  } else {
+    return undefined;
+  }
+}
+
+/**
+ *  @ignore
+ * Helper utility to retrieve `boolean` value from given string,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getBooleanOrUndefined(value: any): boolean | undefined {
+  if (value != undefined) {
+    return value.toString() === "true";
+  } else {
+    return undefined;
+  }
+}
+
+/**
+ *  @ignore
+ * Helper utility to retrieve `JSON` value from given input,
+ * or undefined if not passed in. Treats empty string as invalid JSON
+ * @param value
+ */
+export function getJSONOrUndefined(value: any): number | undefined {
+  if (value == undefined || (typeof value === "string" && !value.toString())) {
+    return undefined;
+  } else {
+    try {
+      return JSON.parse(JSON.stringify(value));
+    } catch (err) {
+      throw new TypeError(
+        `${JSON.stringify(value, undefined, 2)} expected to be in proper JSON format, or undefined`
```

Comment:
The service/entity specific errors are handled in the serializers. This is a utility for handling JSON format data.

Reply:
On second thoughts, I don't think this catch block will catch any error....

If JSON.parse() was called directly on `value`, then yes, a SyntaxError could be thrown.
But, since we are calling JSON.stringify() first and then JSON.parse(), the latter will never be throwing an error.

If you want `getJSONOrUndefined()` to always return either undefined or a proper JSON object representing key/value pair, then the current state of this helper method is not helpful. We can discuss offline if you want on how best to tackle this.